### PR TITLE
Carry all minor changes to dev sql microsoft.sql 2022 11 01 preview

### PR DESCRIPTION
**Pull request by jeremyfrosti**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ericshape** [comment]
@jeremyfrosti, you could also use this command to trigger the pipeline if it is stuck. 

> /azp run

**Comment by raych1** [comment]
@ericshape , check the build log which has the breaking change details. For rendering issue, can you raise the issue in this [tool support channel](https://teams.microsoft.com/l/channel/19%3a9efda948665d481086766c0b7572abc3%40thread.skype/Service%2520API%2520Toolset?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47).

**Comment by jeremyfrosti** [comment]
@raych1 On one of the build issues, we are getting a TypeFormatChange because a double was changed to an int. While this makes sense for the breaking change, it seems that the comparison is to 2021-11-01, and when I check 2022-08-01-preview, this change was already made. I wouldn't expect a breaking change then? Am I missing something here?
Error:
<testcase classname="Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json" name="TypeFormatChanged" file="Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json"> <failure message="| Rule |&#xA;|------|&#xA;|❌ [1023 - TypeFormatChanged](https://github.com/Azure/openapi-diff/tree/master/docs/rules/1023.md)|&#xA;&#xA;| Messges |&#xA;| --------|&#xA;| The new version has a different format than the previous one. <br>New: [Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json](https://github.com/Azure/azure-rest-api-specs/blob/ebf013e3c956ceb3ba8159cd6b1d28b31c861a8c/specification/sql/resource-manager/Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json#L158:9)<br>Old: [Microsoft.Sql/stable/2021-11-01/ManagedDatabaseRestoreDetails.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/sql/resource-manager/Microsoft.Sql/stable/2021-11-01/ManagedDatabaseRestoreDetails.json#L102:9) |&#xA;| The new version has a different format than the previous one. <br>New: [Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json](https://github.com/Azure/azure-rest-api-specs/blob/ebf013e3c956ceb3ba8159cd6b1d28b31c861a8c/specification/sql/resource-manager/Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json#L186:9)<br>Old: [Microsoft.Sql/stable/2021-11-01/ManagedDatabaseRestoreDetails.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/sql/resource-manager/Microsoft.Sql/stable/2021-11-01/ManagedDatabaseRestoreDetails.json#L116:9) |&#xA;" type="ValidationResult"/> </testcase>

2021-11-01: https://github.com/Azure/azure-rest-api-specs/blob/release-sql-Microsoft.Sql-2022-11-01-preview/specification/sql/resource-manager/Microsoft.Sql/stable/2021-11-01/ManagedDatabaseRestoreDetails.json#L102

2022-08-01: https://github.com/Azure/azure-rest-api-specs/blob/release-sql-Microsoft.Sql-2022-11-01-preview/specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/ManagedDatabaseRestoreDetails.json#L159

2022-11-01: https://github.com/Azure/azure-rest-api-specs/blob/release-sql-Microsoft.Sql-2022-11-01-preview/specification/sql/resource-manager/Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json#L159

**Comment by raych1** [comment]
> @raych1 On one of the build issues, we are getting a TypeFormatChange because a double was changed to an int. While this makes sense for the breaking change, it seems that the comparison is to 2021-11-01, and when I check 2022-08-01-preview, this change was already made. I wouldn't expect a breaking change then? Am I missing something here? Error:
> 
> `<testcase classname="Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json" name="TypeFormatChanged" file="Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json"> <failure message="| Rule |&#xA;|------|&#xA;|❌ [1023 - TypeFormatChanged](https://github.com/Azure/openapi-diff/tree/master/docs/rules/1023.md)|&#xA;&#xA;| Messges |&#xA;| --------|&#xA;| The new version has a different format than the previous one. <br>New: [Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json](https://github.com/Azure/azure-rest-api-specs/blob/ebf013e3c956ceb3ba8159cd6b1d28b31c861a8c/specification/sql/resource-manager/Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json#L158:9)<br>Old: [Microsoft.Sql/stable/2021-11-01/ManagedDatabaseRestoreDetails.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/sql/resource-manager/Microsoft.Sql/stable/2021-11-01/ManagedDatabaseRestoreDetails.json#L102:9) |&#xA;| The new version has a different format than the previous one. <br>New: [Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json](https://github.com/Azure/azure-rest-api-specs/blob/ebf013e3c956ceb3ba8159cd6b1d28b31c861a8c/specification/sql/resource-manager/Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json#L186:9)<br>Old: [Microsoft.Sql/stable/2021-11-01/ManagedDatabaseRestoreDetails.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/sql/resource-manager/Microsoft.Sql/stable/2021-11-01/ManagedDatabaseRestoreDetails.json#L116:9) |&#xA;" type="ValidationResult"/> </testcase>`
> 
> 2021-11-01: https://github.com/Azure/azure-rest-api-specs/blob/release-sql-Microsoft.Sql-2022-11-01-preview/specification/sql/resource-manager/Microsoft.Sql/stable/2021-11-01/ManagedDatabaseRestoreDetails.json#L102
> 
> 2022-08-01: https://github.com/Azure/azure-rest-api-specs/blob/release-sql-Microsoft.Sql-2022-11-01-preview/specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/ManagedDatabaseRestoreDetails.json#L159
> 
> 2022-11-01: https://github.com/Azure/azure-rest-api-specs/blob/release-sql-Microsoft.Sql-2022-11-01-preview/specification/sql/resource-manager/Microsoft.Sql/preview/2022-11-01-preview/ManagedDatabaseRestoreDetails.json#L159

@jeremyfrosti , there're other breaking changes reported and needs the breaking change board review.

**Review by ericshape — APPROVED** [review]

**Comment by jeremyfrosti** [comment]
@raych1 Could you give this PR a review now? We have resolved the breaking change issue. These changes are mostly just description changes and some example file updates.

**Comment by jeremyfrosti** [comment]
@raych1 Gentle ping on this

**Comment by jeremyfrosti** [comment]
@raych1 Gentle ping on this

File: specification/sql/resource-manager/Microsoft.Sql/preview/2022-11-01-preview/LongTermRetentionPolicies.json [hunk 1]
```diff
@@ -186,6 +186,29 @@
     }
   },
   "definitions": {
+    "BaseLongTermRetentionPolicyProperties": {
```
Comment (raych1) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2022-11-01-preview/LongTermRetentionPolicies.json:189]: model name change might cause SDK breaking change.

**Comment by raych1** [comment]
@weidongxu-microsoft @msyyc @tadelesh , can you review the potential SDK breaking change?

**Comment by tadelesh** [comment]
@raych1 It's hard to distinguish all the breakings for such composite tag. I'll approve the breaking since it is a preview version, but we need to figure out what is the best way to deal with such PR.

**Review by raych1 — APPROVED** [review]

**Review by ihsanali01 — COMMENTED** [review]
Help me to make it better.

**Pull request merged**